core/modules/entity/entity.class.inc

```diff
@@ -278,6 +319,34 @@ abstract class Entity extends stdClass implements EntityInterface {
     }
     return $values;
   }
+
+  /**
+   * Implements Backdrop\entity\EntityInterface::getRevisionId().
+   */
+  public function getRevisionId() {
+    return NULL;
+  }
+
+  /**
+   * Implements Backdrop\entity\EntityInterface::isActiveRevision().
+   */
+  public function isActiveRevision() {
+    return $this->is_active_revision;
+  }
+
+  /**
+   * Implements Backdrop\entity\EntityInterface::getActiveRevision().
+   */
+  public function getActiveRevision() {
+    return $this->active_revision;
+  }
+
+  /**
+   * Implements Backdrop\entity\EntityInterface::setActiveRevision().
+   */
+  public function setActiveRevision() {
+    $this->is_active_revision = TRUE;
```

Comment:
If we're updating `is_active_revision`, I think we should set `active_revision_id` to be the `$this->getRevisionId()` so that they match. In addition to that, I think that if `is_active_revision` and the revision ID do not match up when saving the entity, an exception should be thrown. For example this should not be allowed to save:

```
$node_revision = node_load(NULL, 10);
$node->setActiveRevision();
$node->active_revision_id = 9;
$node->save();
```

Because `is_active_revision` does not agree with the `vid` and `active_revision_id` -- both of which should be 10 at this point.

Reply:
But `ENTITY->active_revision_id` would be read-only so that code block wouldnt have an effect. 

However as it is now though, the `ENTITY->active_revision_id` property and the `getActiveRevisionId()` are both unused in the code, just added as utilities for UX. We could drop both of those if we think it isnt necessary, and skip having this discussion.

Aside, should `setActiveRevision()` really be `setAsActiveRevision()`?

Reply:
> Aside, should setActiveRevision() really be setAsActiveRevision()?

Maybe `setIsActiveRevision()` would be the best alignment.

> But ENTITY->active_revision_id would be read-only so that code block wouldnt have an effect.

PHP doesn't have read-only properties, so you _could_ write to it even if you weren't supposed to. The only way to prevent writing is to it private and only accessible via `getActiveRevisionId()`.

But yeah, if we're not using it in the first place, perhaps save it for later.